### PR TITLE
Sitemap Plugin

### LINKED ISSUE #52: Generate a sitemap
Since I saw blogging support is planned, I thought I'd create a request for sitemap support as well (and maybe even RSS feed while we're at it?). Could try and work on it as well.

### PR DESCRIPTION
**Pull request by ekoeryanto**
**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot: 

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
fix #52 
ping #158 

You have tested in the following browsers: (Providing a detailed version will be better.)

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: packages/@vuepress/plugin-sitemap/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+module.exports = ({ hostname, changefreq = 'daily', cacheTime = 600000, urls = [], ...others }, context) => ({
+  async generated () {
+    if (!hostname) {
+      return logger.warn(`\nNot generating sitemap because required 'hostname' option doesn't exist `)
+    }
+
+    logger.wait('\nGenerating sitemap...')
+
+    const { pages } = context.getSiteData()
```
[the remark this answers is not in the document]
Comment (ekoeryanto) [reply packages/@vuepress/plugin-sitemap/index.js]: wait, I'll check the `context.pages`

[the remark this answers is not in the document]
Comment (ekoeryanto) [reply packages/@vuepress/plugin-sitemap/index.js]: I missed that context, I just have idea to create feed plugin with that

**Comment by ekoeryanto** [comment]
@ulivz anyway to get generated html string to `context.pages` ?

**Comment by ulivz** [comment]
Thanks, But you can maintain this plugin by yourself. we don't plan to add all plugins to the official repo.

**Pull request closed without merging**